src/autoscaler/api/policyvalidator/policy_validator.go

```diff
@@ -0,0 +1,369 @@
+	}
+
+	// check InstanceMinCount, IntanceMaxCount and InitialInstanceMinCount for schedules
+	if policy.Schedules == nil {
+		return
+	}
+	schedulesContext := gojsonschema.NewJsonContext("schedules", rootContext)
+
+	pv.validateRecurringSchedules(policy, schedulesContext, result)
+	pv.validateSpecificDateSchedules(policy, schedulesContext, result)
+}
+
+func (pv *PolicyValidator) validateRecurringSchedules(policy *models.ScalingPolicy, schedulesContext *gojsonschema.JsonContext, result *gojsonschema.Result) {
+	recurringScheduleContext := gojsonschema.NewJsonContext("recurring_schedule", schedulesContext)
+	for i, recSched := range policy.Schedules.RecurringSchedules {
+		if recSched.ScheduledInstanceMin >= recSched.ScheduledInstanceMax {
+			instanceMinContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d.instance_min_count", i), recurringScheduleContext)
+			errDetails := gojsonschema.ErrorDetails{
+				"instance_min_count": recSched.ScheduledInstanceMin,
+				"instance_max_count": recSched.ScheduledInstanceMin,
+			}
+			formatString := "instance_min_count {{.instance_min_count}} is higher or equal to instance_max_count {{.instance_max_count}}"
+			err := newPolicyValidationError(instanceMinContext, formatString, errDetails)
+			result.AddError(err, errDetails)
+		}
+
+		if recSched.ScheduledInstanceInit > recSched.ScheduledInstanceMax {
+			initialInstanceMinContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d.initial_min_instance_count", i), recurringScheduleContext)
+			errDetails := gojsonschema.ErrorDetails{
+				"initial_min_instance_count": recSched.ScheduledInstanceInit,
+				"instance_max_count":         recSched.ScheduledInstanceMin,
+			}
+			formatString := "initial_min_instance_count {{.initial_min_instance_count}} is greater than instance_max_count {{.instance_max_count}}"
+			err := newPolicyValidationError(initialInstanceMinContext, formatString, errDetails)
+			result.AddError(err, errDetails)
+		}
+
+		//start_time should be before end_time
+		if compareTimesGTEQ(recSched.StartTime, recSched.EndTime) {
+			currentRecSchedContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d", i), recurringScheduleContext)
+			errDetails := gojsonschema.ErrorDetails{
+				"i": i,
+			}
+			formatString := "start_time is after end_time"
+			err := newPolicyValidationError(currentRecSchedContext, formatString, errDetails)
+			result.AddError(err, errDetails)
+		}
+		// start_date should be after current_date and before end_date
+		var startDate, endDate time.Time
+		location, _ := time.LoadLocation(policy.Schedules.Timezone)
+
+		if recSched.StartDate != "" {
+			startDate, _ = time.ParseInLocation(DateLayout, recSched.StartDate, location)
+
+			if startDate.Sub(time.Now()) < 0 {
+				currentRecSchedContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d", i), recurringScheduleContext)
+				errDetails := gojsonschema.ErrorDetails{
+					"i": i,
+				}
+				formatString := "start_date is before current_date"
```

Comment:
There is no "hh:mm:ss" setting on start date, so time.ParseInLocation will use 00:00:00 when converts startDate.  With the implementation above, I can't define a schedule starting from today .. 
for example, assuming I want to create a recurring schedule for 10:00 ~ 18:00 with start date  `2019-03-01` at **2019-03-01 8:00AM  **,   the first trigger should be 2019-03-01 10:00AM which is later than the CURRENT time.  This is a correct schedule definition.  But in above implementation,  the start date is 2019-03-01 00:00AM which is before CURRENT ,  the policy is rejected.

Reply:
fixed.